src/power_grid_model/core/power_grid_dataset.py

```diff
@@ -466,22 +466,15 @@ def _filter_attributes(self, attributes):
         for key in keys_to_remove:
             del attributes[key]
 
-    def _filter_with_option(self):
-        if self._data_filter is ComponentAttributeFilterOptions.RELEVANT:
-            for attributes in self._data.values():
-                self._filter_attributes(attributes)
-
     def _filter_with_mapping(self):
         for component_type, attributes in self._data.items():
             if component_type in self._data_filter:
                 filter_option = self._data_filter[component_type]
-                if filter_option is ComponentAttributeFilterOptions.RELEVANT:
+                if filter_option is ComponentAttributeFilterOptions.relevant:
                     self._filter_attributes(attributes)
 
     def _post_filtering(self):
-        if isinstance(self._data_filter, ComponentAttributeFilterOptions):
-            self._filter_with_option()
-        elif isinstance(self._data_filter, dict):
+        if isinstance(self._data_filter, dict):
             self._filter_with_mapping()
```

Comment:
This is removed due to the fact that `self._data_filter` is a processed version that is always a aggregated version of input `data_filter`.